### ISSUE TITLE
Support nested assumes

### ISSUE THREAD

**Issue by yanksyoon** [juju/python-libjuju#938]
### Description

There is an error when using juju lib with a charm that has nested [`assumes`](https://juju.is/docs/sdk/metadata-yaml#heading--assumes) statement.

Currently, the library seems to only [support 1 level of nesting](https://github.com/juju/python-libjuju/blob/3edef16a7e1d2d1fb8b303c71c03a8ed46f022cf/juju/client/facade.py#L683-L695) whereas by [definition](https://github.com/juju/charm/blob/master/assumes/parser.go) the n-depth nesting statements are supported. For example, the [mysql-k8s charm](https://github.com/canonical/mysql-k8s-operator/blob/main/metadata.yaml#L74-L82) uses a nesting of depth 2.

### Urgency

Annoying bug in our test suite

### Python-libjuju version

all versions

### Juju version

all versions

### Reproduce / Test

```python
import pytest
import OpsTest
from juju.model import Model
from pytest_operator.plugin import OpsTest

@pytest.fixture(scope="module")
def model(ops_test: OpsTest) -> Model:
    model = ops_test.model
    assert model
    return model

def test_deploy_nested_assumes(model: Model):
    await model.deploy("mysql-k8s", channel="8.0/edge", trust=True)
    await model.wait_for_idle(status="active", apps=["mysql-k8s"])
```
```

**Comment by cderici** [comment]
@carlcsaposs-canonical I was under the impression that you wanted to have the fix in a release that supports both 2.9 and 3.x at the same time, which is still under assessment. If you *can* use the `2.9.44` then I can happily fix this there and maybe we can even get a quick patch release (`2.9.44.1`) with this for you guys no problem 👍

**Comment by cderici** [comment]
@carlcsaposs-canonical That, we can do for sure, I wasn't aware that this was blocking your release. I'll prioritize it for you guys 👍
